egnoramoose.py

```diff
@@ -1,119 +1,167 @@
 class Pawn:
-	def __init__(self,location):
-		self.location = location
+    pass
+
+
+class Location:
+    def __init__(self, *location):
+        self.row = -1
+        self.column = -1
+        self._convert(location)
+
+    def __sub__(self, other):
+        return Location(self.row-other.row, self.column-other.column)
+
+    def __str__(self):
+        return '{}{}'.format(self.row, self.column)
+
+    @staticmethod
+    def _list_to_int(list_obj):
+        return [int(i) for i in list_obj]
+
+    def _convert(self, location):
+        if len(location) == 1:
+            location_str = location[0]
+            assert len(location_str) == 2
+            self.row, self.column = Location._list_to_int(location_str)
+        elif len(location) == 2:
+            for arg in location:
+                if isinstance(arg, str):
+                    assert len(arg) == 1
+            self.row, self.column = Location._list_to_int(location)
+
+    @property
+    def location(self):
+        return (self.row, self.column)
```

Comment:
When ever someone instantiate our object and do location_object.location, this function triggers. Thats what @property is for. As you see, I don't have a location variable. However, you could be the @property location will trigger not the location instance variable when a user does location_object.location